src/hb-face.cc

```diff
@@ -278,13 +283,13 @@ hb_face_destroy (hb_face_t *face)
 
 /**
  * hb_face_set_user_data: (skip)
- * @face: a face.
- * @key:
- * @data:
- * @destroy:
- * @replace:
- *
+ * @face: A face object
+ * @key: The user-data key
+ * @data: A pointer to the user data
+ * @destroy: A callback to call when @data is not needed anymore
```

Comment:
@destroy can be null.  Need both g-i annotations and maybe doc?  Donno.

Reply:
Can do!

Reply:
Are all destroy's nullable, no exceptions?

Also, looking at the GI annotations, it sounds like (optional) might be a better annotation for this than (nullable), but IDK.

Reply:
Yes, all are optional. I never understood the difference between optional and nullable. What's your understanding?

Reply:
It seems to me that they want 'optional' to mean the client/user chooses not to use the parameter, but 'nullable' might be something that's required but sometimes has NULL as a valid value?

Reply:
(And that may be a bigger difference on output params than on input, not really mattering here)